tendrl/node_agent/manager/rpc.py

```diff
@@ -23,48 +25,64 @@ def __init__(self):
         self.client = etcd.Client(**etcd_kwargs)
         self.bridge_id = str(uuid.uuid4())
 
+    def _process_job(self, raw_job, job_key):
+        # Pick up the "new" job that is not locked by any other bridge
+        if raw_job['status'] == "new" and raw_job["sds_type"] == "generic":
+                raw_job['status'] = "processing"
+                # Generate a request ID for tracking this job
+                # further by tendrl-api
+                req_id = str(uuid.uuid4())
+                raw_job['request_id'] = "%s/flow_%s" % (
+                    self.bridge_id, req_id)
+                self.client.write(job_key, json.dumps(raw_job))
+                LOG.info("Processing API-JOB %s" % raw_job[
+                    'request_id'])
+                try:
+                    result, err = self.invoke_flow(
+                        raw_job['flow'], raw_job
+                    )
+                except FlowExecutionFailedError as e:
+                    LOG.error(e)
+                    raise
+
+                raw_job['status'] = "finished"
+                raw_job["response"] = {
+                    "result": result,
+                    "error": err
+                }
+                return raw_job, True
+        else:
+            return raw_job, False
+
     def _acceptor(self):
         while True:
             jobs = self.client.read("/api_job_queue")
+            gevent.sleep(2)
```

Comment:
This reading from the job queue is done through greenlets(kind of light-wait threads). These threads have to make way for other threads by sleeping periodically. Else the other threads starve. Here job-queue executing greenlet sleeps for 2 secs and wakes up and checks job queue for new jobs in a continuous loop.
Earlier when the sleep was inside, the sleep was not getting called once all the new jobs in the queue are processed, because of this other greenlets were starved. To fix that it was moved out